src/Data/ArrayBuffer/DataView.js

```diff
@@ -3,43 +3,35 @@
 // module Data.ArrayBuffer.DataView
 
 
-exports.whole = function(b) {
-  return new DataView(b);
-}
-
-exports.sliceImpl = function(just, nothing, s, l, b) {
-  return ((s + l)>>>0) <= b.byteLength ? just(new DataView(b, s, l)) : nothing;
-}
-
-exports.buffer = function(v) {
-  return v.buffer;
-}
-
-exports.byteOffset = function(v) {
-  return v.byteOffset;
-}
-
-exports.byteLength = function(v) {
-  return v.byteLength;
-}
-
-exports.getterImpl = function(just, nothing, s, l, e, v, o) {
-  return function() {
-    return ((o + l)>>>0) <= v.byteLength? just(v[s].call(v,o,e)) : nothing;
-  };
-}
-
-exports.setter = function(s) {
-  return function(e) {
-    return function(v) {
-      var f = v[s];
-      return function(n) {
-        return function(o) {
-            return function() {
-            f.call(v,o,n,e);
-          };
-        };
-      };
-    };
-  };
-}
+exports.whole = function whole (b) {
+    return new DataView(b);
+};
+
+exports.remainderImpl = function remainderImpl (b,i) {
+    return new DataView(b,i);
+};
+
+exports.partImpl = function partImpl (b,i,j) {
+    return new DataView(b,i,j);
+};
+
+exports.buffer = function buffer (v) {
+    return v.buffer;
+};
+
+exports.byteOffset = function byteOffset (v) {
+    return v.byteOffset;
+};
+
+exports.byteLength = function byteLength (v) {
+    return v.byteLength;
+};
+
+exports.getterImpl = function getterImpl (data, s, l, e, v, o) {
+    return ((o + 1) >>> 0) <= v.byteLength ? data.just (v[s].call(v,o,e)) : data.nothing;
+};
+
+exports.setterImpl = function setterImpl (s,e,v,n,o) {
```

Comment:
I don't know why this didn't perform range-checking in the same way `getterImpl` does... the previous version was probably broken, we could fix it here now...

Reply:
I think so, that's what -arrays does:

https://pursuit.purescript.org/packages/purescript-arrays/5.2.0/docs/Data.Array.ST#v:poke

Reply:
Alright, I've got this implemented, at least I think how it should be.